app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -31,101 +31,160 @@ import androidx.compose.material3.ExperimentalMaterial3Api
-    appState: NiaAppState = rememberNiaAppState(windowSizeClass)
+    networkMonitor: NetworkMonitor,
+    appState: NiaAppState = rememberNiaAppState(
+        networkMonitor = networkMonitor,
+        windowSizeClass = windowSizeClass
+    ),
 ) {
-    NiaTheme {
-        val background: @Composable (@Composable () -> Unit) -> Unit =
-            when (appState.currentDestination?.route) {
-                ForYouDestination.route -> { content -> NiaGradientBackground(content = content) }
-                else -> { content -> NiaBackground(content = content) }
-            }
+    val background: @Composable (@Composable () -> Unit) -> Unit =
+        when (appState.currentDestination?.route) {
+            ForYouDestination.route -> { content -> NiaGradientBackground(content = content) }
+            else -> { content -> NiaBackground(content = content) }
+        }
 
-        background {
-            Scaffold(
-                modifier = Modifier.semantics {
-                    testTagsAsResourceId = true
-                },
-                containerColor = Color.Transparent,
-                contentColor = MaterialTheme.colorScheme.onBackground,
-                contentWindowInsets = WindowInsets(0, 0, 0, 0),
-                bottomBar = {
-                    if (appState.shouldShowBottomBar) {
-                        NiaBottomBar(
-                            destinations = appState.topLevelDestinations,
-                            onNavigateToDestination = appState::navigate,
-                            currentDestination = appState.currentDestination
-                        )
-                    }
+    background {
+
+        val snackbarHostState = remember { SnackbarHostState() }
+
+        Scaffold(
+            modifier = Modifier.semantics {
+                testTagsAsResourceId = true
+            },
+            containerColor = Color.Transparent,
+            contentColor = MaterialTheme.colorScheme.onBackground,
+            contentWindowInsets = WindowInsets(0, 0, 0, 0),
+            snackbarHost = { SnackbarHost(snackbarHostState) },
+            topBar = {
+                // Show the top app bar on top level destinations.
+                val topLevelDestination =
+                    appState.topLevelDestinations[appState.currentDestination?.route]
+                if (topLevelDestination != null) {
+                    NiaTopAppBar(
+                        titleRes = topLevelDestination.titleTextId,
+                        actionIcon = NiaIcons.Settings,
+                        actionIconContentDescription = stringResource(
+                            id = settingsR.string.top_app_bar_action_icon_description
+                        ),
+                        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+                            containerColor = Color.Transparent
+                        ),
+                        onActionClick = { appState.toggleSettingsDialog(true) }
```

Comment:
If you're toggling the value, you don't need to pass the new value in, do you? if not, call the method `setSettingsDialog(true)`?

Reply:
Good spot. There's a few other instances of this in our codebase. Bug filed: https://github.com/android/nowinandroid/issues/351